File: python/exercism/bools/triangle.py
def triangle(sides):
    a, b, c = sides
    if a == 0 or b == 0 or c == 0:
        return False
    return a + b >= c and b + c >= a and a + c >= b


def equilateral(sides):
    if not triangle(sides):
        return False
    return len(set(sides)) == 1


def isosceles(sides):
    if not triangle(sides):
        return False
    a, b, c = sides
    return len(set(sides)) < 3


def scalene(sides):
    if not triangle(sides):
        return False
    return len(set(sides)) == 3

File: ruby/exercism/floating_point_numbers/triangle.rb
class Triangle
  def initialize(sizes)
    @a, @b, @c = sizes
  end

  def triangle?
    return false unless @a.positive? && @b.positive? && @c.positive?

    @a <= (@b + @c) && @b <= (@a + @c) && @c <= (@a + @b)
  end

  def equilateral?
    return false unless triangle?

    (@a == @b) && (@b == @c)
  end

  def isosceles?
    return false unless triangle?

    (@a == @b) || (@a == @c) || (@c == @b)
  end

  def scalene?
    return false unless triangle?

    (@a != @b) && (@a != @c) && (@b != @c)
  end
end
